HealthcareSystem/Backend/Service/ActionBenefitService/RabbitMqActionBenefitService/RabbitMqActionBenefitService.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Backend.Model.Pharmacies;
+using Backend.Service.Pharmacies;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Backend.Service
+{
+    public class RabbitMqActionBenefitMessageingService : BackgroundService

```

Comment:
I would maybe make an interface for this as well, something like IActionBenefitMessageingService? Just to to make the rest of the system a bit less tied to RabbitMq (like whatever part of the system will call Unsubscribe / Subscribe, I assume it will be the pharmacy service or something - it would also be useful for testing that part of the system). Also, if this will need to be injected anywhere it needs an interface.

Reply:
Ok, will make interface. But its not needed for injection, tried it in another project when i was researching rabbitmq and it worked.